### DISCUSSION TITLE
TSML Version 3.14.8: Map Issue

### DISCUSSION THREAD

**Discussion by ebart455987** [code4recovery/12-step-meeting-list#1014] [General]
### Contact Details

wm@fortworthaa.org

### Website With Issue

www.fortworthaa.org

### What happened?

TSML Version 3.14.8: during adding or updating a meeting the map does not update/doesn't display.  Also the Update/Publish buttons are not available if address is entered/changed.  The Update/Publish buttons appear when any other changes are entered, (e.g meeting type, time). We're using Wordpress 2013 Theme.  https://www.fortworthaa.org/meetings/

### TSML Version

Latest (default)

### Wordpress Version

Latest (default)

### What browsers are you seeing the problem on?

Safari

**Comment by tim-rohrer** [comment]
Hi Bart,

We ask users to open Discussions vice Issues to give us a chance to investigate before putting it on the developer plate :-)

As 3.14.8 was recently released, it is possible we introduced a bug, but I'm not currently able to reproduce your issue, so let's run check the common causes first (you'll also find several threads on this in the Discussions).

A common cause is that something has happened to your mapping key, usually provided by MapBox but it could be Google if you configured your site that way. Look into that, and let us know what happing service you're trying to use.

Another common issue is geocoding. With the latest release we pulled the beta geocoding service option and defaulted everyone back to legacy. This appears to be working, so if this is an issue on your site it should show in the console logs. Open debugging on your browser and watch while you enter an address and tab off. You should see successful geocoding reported. 

If not, please paste the error message here.

Tim

**Comment by joshreisner** [comment]
hi, looks like you’re using a custom google maps key. could you check your google api console and confirm that geocoding is turned on for your key?

**Reply by joshreisner** [reply]
hi @ebart455987 to reply to your reply below, you log in to the [Google Cloud Console](https://console.cloud.google.com), make sure the correct project is selected, and then click on Sidebar > APIs & Services > Enabled APIs & Services, and confirm that the following are enabled:

* Geocoding API
* Google Maps Javascript API

![google-cloud-console](https://user-images.githubusercontent.com/1551689/209504404-91ec7ddd-5959-46df-9ac3-8a6c77100f2f.png)

**Reply by joshreisner** [reply]
When geocoding on your site I see the message `This API project is not authorized to use this API.`so I'm pretty sure this is the problem!

**Comment by ebart455987** [comment]
It started after upgrading to latest version of TSML.

Where do I find the geocoding setting?

Have a great day!
Eric

**Comment by ebart455987** [comment]
Hi, I found the GOOGLE api key page.
I tried regenerating a new key, didn’t work.
From the looks of this page it appears the api key is enabled.
[Screenshot 2022-12-25 at 10.29.00 PM.png]

Yours in Service,
Eric B~
Web Servant
www.fortworhaa.org




On Dec 25, 2022, at 11:28 AM, Josh Reisner ***@***.***> wrote:



hi, looks like you’re using a custom google maps key. could you check your google api console and confirm that geocoding is turned on for your key?

**Comment by ebart455987** [comment]
Josh,

That worked!  The Geocoding API was missing.

Thank You!


Yours in Service,
Eric B~
Web Servant
www.fortworhaa.org



Have a great day!

On Dec 26, 2022, at 12:18 AM, Josh Reisner ***@***.***> wrote:



hi @ebart455987<https://github.com/ebart455987> to reply to your reply below, you log in to the Google Cloud Console<https://console.cloud.google.com/>, make sure the correct project is selected, and then click on Sidebar > APIs & Services > Enabled APIs & Services, and confirm that the following are enabled:

  *   Geocoding API
  *   Google Maps Javascript API

[google-cloud-console]<https://user-images.githubusercontent.com/1551689/209504404-91ec7ddd-5959-46df-9ac3-8a6c77100f2f.png>

**Comment by ebart455987** [comment]
Josh,

I’m no savvy about API Keys.  Is it best to use a certain API Key?  I don’t remember how I ended up with a custom key.

Also, For some reason the Google API Key is under my personal Google Account as opposed to the Central Office.  


Have a great day!
Eric

**Reply by tim-rohrer** [reply]
A simpler setup is to use a MapBox key, and then our Google key is used for geocoding. 

But if this is working for you, great!

**Reply by ebart455987** [reply]
If it's simpler, I'd like to try it.  Where do I get a MapBox key?  And where is "our Google key" ?  

<img width="451" alt="Screenshot 2022-12-26 at 11 50 21 AM" src="https://user-images.githubusercontent.com/55007577/209569316-0028cbe7-a240-464d-91ca-9f1e29a22079.png">

**Reply by ebart455987** [reply]
I figured out how to get a Mapbox key.  I made one for FTWCO, used it instead of a public version.  I entered the FTWCO api Mapbox key into the Maapbox Access Token field in TSML Settings. 
The TSML now uses Mapbox on our site.  Seems to work fine. 
Now all I think I need is "our Google Key"
<img width="748" alt="Screenshot 2022-12-26 at 12 44 53 PM" src="https://user-images.githubusercontent.com/55007577/209572654-9ab3dca1-a40e-4086-b37f-1e1f4131f4f2.png"> 
<img width="451" alt="Screenshot 2022-12-26 at 12 47 40 PM" src="https://user-images.githubusercontent.com/55007577/209572822-3ef90c9f-1743-41f1-8a75-2593fa965b7b.png">

**Reply by tim-rohrer** [reply]
While using Mapbox, leave the Google API key field empty. You don't need it.
